### DISCUSSION TITLE
How to use dates properly?

### DISCUSSION THREAD

**Discussion by shervinkoushan** [patrykandpatrick/vico#943] [Questions] [answered]
### Question

I want to display dates on the x-axis. Previously I have used MPAndroidChart to achieve this.
The x-values have then been milliseconds since epoch (though seconds would have sufficed).
However, when I use Vico the graph is just a straight line when the time diff between the values is large.

<img width="390" alt="Screenshot 2024-11-03 at 09 52 49" src="https://github.com/user-attachments/assets/06f50351-f34b-46cd-ac22-f94984063a2b">

Note that the desired behavior also considers the distance in time between points, i.e., points that are close in time should also be close in the graph (there are three points in the below example, taken from the existing app).

![Screenshot 2024-11-03 at 09 53 50](https://github.com/user-attachments/assets/d303ffcc-badb-4a0f-9f2a-c8b18423872b)

I took a look here but was not able to achieve the desired outcome by following these examples: https://patrykandpatrick.com/vico/wiki/cartesian-charts/value-formatting/

### Vico version(s)

2.0.0-beta.2

### UI framework(s)

Jetpack Compose

**Comment by Tyler-Lopez** [comment, marked as answer]
It looks like the problem here is that your chart is overly zoomed in, does that sound right? Have you considered setting a maximum zoom or just restricting the zoom level to always be completely zoomed out?

I could be missing some here though - as it isn't clear to me why you are seeing a straight line if not for the zoom level being so zoomed-in that very gradual changes appear as straight.

**Comment by shervinkoushan** [comment]
I was using
` 
zoomState = rememberVicoZoomState()
`


Changing it to         
`zoomState = rememberVicoZoomState(initialZoom = Zoom.Content)
`
I can see the whole graph:
<img width="389" alt="Screenshot 2024-11-05 at 20 54 37" src="https://github.com/user-attachments/assets/e648b90f-1005-48e3-824a-138e44846bf0">
Thanks!

Setting the itemPlacer to 
`remember { HorizontalAxis.ItemPlacer.aligned(spacing = 1000) }
`
The bottom labels look more sensible:
<img width="383" alt="Screenshot 2024-11-05 at 20 52 16" src="https://github.com/user-attachments/assets/ed32cd16-836d-49dc-88a3-21d0267a8462">

But I wonder what the best way to choose the spacing would be?

**Reply by Tyler-Lopez** [reply]
I think maybe the easiest solution would be to set your spacing to the unix seconds in a day perhaps? Alternatively, you can implement your own `HorizontalAxis.ItemPlacer` which overrides the functions to provide "x labels" to only provide the exact x-values you care about.

**Reply by shervinkoushan** [reply]
Thanks again. I believe implementing my own `HorizontalAxis.ItemPlacer` will be the best solution in this case.

**Reply by Gowsky** [reply]
Hello, @shervinkoushan. If you want one label per day, you don't need a custom `HorizontalAxis.ItemPlacer`. It's enough to set the `xStep` to one day using `getXStep` parameter of `rememberCartesianChart`.

**Reply by shervinkoushan** [reply]
Thanks for the input @Gowsky! This seems to be exactly what I want (the range shown in the graph will be from 24 hours to several years. If I adjust this parameter based on my data, it should be straightforward to achieve the look I am after).
